git-protocol/src/fetch.rs

```diff
@@ -433,32 +433,42 @@ where
     R: AsyncRead + Unpin + Send + 'static,
     W: AsyncWrite + Unpin + Send + 'static,
 {
+    use git_packetline::encode::flush_to_write;
     use git_transport::{
         client::git::{ConnectMode, Connection},
         Protocol,
     };
 
     let stop = AtomicBool::new(false);
-    let conn = Connection::new(
-        recv,
-        send,
-        Protocol::V2,
-        opt.repo.clone(),
-        None::<(String, Option<u16>)>,
-        ConnectMode::Daemon,
-    );
-    let receive_pack = build_receive_pack(&stop);
-    let task = blocking::unblock(move || {
-        let mut delegate = Fetch::new(opt, filter_refs, receive_pack);
-        future::block_on(git_protocol::fetch::fetch(
-            conn,
-            &mut delegate,
-            |_| unreachable!("credentials helper requested"),
-            progress::Discard,
-        ))
-        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-
-        Ok(delegate.out)
+    let task = blocking::unblock({
+        let mut conn = Connection::new(
+            recv,
+            send,
+            Protocol::V2,
+            opt.repo.clone(),
+            None::<(String, Option<u16>)>,
+            ConnectMode::Daemon,
+        );
+        let receive_pack = build_receive_pack(&stop);
+
+        move || {
+            let mut delegate = Fetch::new(opt, filter_refs, receive_pack);
+            future::block_on(async {
+                let res = git_protocol::fetch::fetch(
+                    &mut conn,
+                    &mut delegate,
+                    |_| unreachable!("credentials helper requested"),
+                    progress::Discard,
+                )
+                .await
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e));
+                flush_to_write(conn.into_inner().1).await.ok();
```

Comment:
The delegate will now send a flush packet at the end of a successful operation, which probably leads to issues now as it would try to flush again in this line. 

If control over this should remain here, there is a [delegate method to override](https://github.com/Byron/gitoxide/blob/928f75ad939e35a159d3d2751d5d0f9d00d796af/git-protocol/src/fetch/delegate.rs#L122-L131).